components/scream/src/physics/cld_fraction/atmosphere_cldfraction.cpp

```diff
@@ -0,0 +1,156 @@
+#include "atmosphere_cldfraction.hpp"
+
+#include "ekat/ekat_assert.hpp"
+#include "ekat/util/ekat_units.hpp"
+
+#include <array>
+
+namespace scream
+{
+  using namespace cldfrac;
+// =========================================================================================
+CldFraction::CldFraction (const ekat::Comm& comm, const ekat::ParameterList& params)
+ : m_cldfraction_comm (comm)
+ , m_cldfraction_params (params)
+{
+  // Nothing to do here
+}
+
+// =========================================================================================
+void CldFraction::set_grids(const std::shared_ptr<const GridsManager> grids_manager)
+{
+  using namespace ekat::units;
+
+  // The units of mixing ratio Q are technically non-dimensional.
+  // Nevertheless, for output reasons, we like to see 'kg/kg'.
+  auto Q = kg/kg;
+  Q.set_string("kg/kg");
+  Units nondim(0,0,0,0,0,0,0);
+
+  const auto& grid_name = m_cldfraction_params.get<std::string>("Grid");
+  auto grid  = grids_manager->get_grid(grid_name);
+  m_num_cols = grid->get_num_local_dofs(); // Number of columns on this rank
+  m_num_levs = grid->get_num_vertical_levels();  // Number of levels per column
+  m_nk_pack  = ekat::npack<Spack>(m_num_levs);
+
+  // Define the different field layouts that will be used for this process
+  using namespace ShortFieldTagsNames;
+
+  // Layout for 3D (2d horiz X 1d vertical) variable defined at mid-level and interfaces 
+  FieldLayout scalar3d_layout_mid { {COL,LEV}, {m_num_cols,m_num_levs} };
+
+  // Set of fields used strictly as input
+  m_required_fields.emplace("qi",   scalar3d_layout_mid, Q,      grid_name);
+  m_required_fields.emplace("alst", scalar3d_layout_mid, nondim, grid_name);
+
+  // Set of fields used strictly as output
+  m_computed_fields.emplace("ast",   scalar3d_layout_mid, nondim, grid_name);
+  m_computed_fields.emplace("aist",  scalar3d_layout_mid, nondim, grid_name);
+
+  // Set of fields used as input and output
+
+}
+
+// =========================================================================================
+void CldFraction::initialize_impl (const util::TimeStamp& t0)
+{
+  m_current_ts = t0;
+}
+
+// =========================================================================================
+void CldFraction::run_impl (const Real dt)
+{
+/* Luca, do we need any of the following code?
```

Comment:
@bartgol , I left a question in the code here for you.  This is copied from other atm. processes, but I'm not sure I understand what is happening here and if its needed for this code.

Reply:
No. I'm assuming you copied that from P3. These lines were for when the interface called f90 stuff. We can purge these lines (also from P3 and shoc).